### PR TITLE
Cduranti/major version

### PR DESCRIPTION
**Pull request by rozap**
* this is split up into a few commits
* remove the (ok, result) pattern in favor of using exceptions for control flow
* parallel chunked uploads

### CODE REVIEW

File: test/output_schema_test.py [hunk 1]
```diff
@@ -337,50 +308,6 @@ def test_change_existing_schema(self):
...
 
-        (ok, result) = get("https://{domain}/id/{fourfour}".format(domain = auth.domain, fourfour = rev.attributes['fourfour']), auth)
-        assert ok, result
+        result = get("https://{domain}/id/{fourfour}".format(domain = auth.domain, fourfour = rev.attributes['fourfour']), auth)
 
         self.assertEqual(result, [{'a': '1', 'd': '6'}, {'a': '2', 'd': '7'}, {'a': '3', 'd': '8'}, {'a': '4', 'd': '9'}])
-
-
-    # Broken due to ingress strategy flags - if we leave this long enough maybe
-    # the OBE will be gone and we can just delete it
```
Comment (branweb1) [inline test/output_schema_test.py]: it's great to see a plan sorta kinda come to fruition

Comment (rozap) [reply test/output_schema_test.py]: wooooo

File: socrata/sources.py [hunk 2]
```diff
@@ -89,6 +90,28 @@ def __next__(self):
...
+        self.done = False
+
+    def read(self, how_much):
+        if self.done:
+            return None
+
+        buf = b''
+        while len(buf) < how_much:
+            try:
+                buf = buf + next(self.gen)
```
Comment (rjmac) [inline socrata/sources.py]: O(n²) alert! :)

Comment (rozap) [reply socrata/sources.py]: gah, `len` is O(n) :(

Comment (rozap) [reply socrata/sources.py]: ahhh yep

**Pull request merged**